python/ray/ml/predictors/integrations/sklearn/sklearn_predictor.py

```diff
@@ -100,4 +124,24 @@ def predict(
             pd.DataFrame: Prediction result.
 
         """
-        raise NotImplementedError
+        register_ray()
+
+        if self.preprocessor:
+            data = self.preprocessor.transform_batch(data)
+
+        if num_estimator_cpus:
+            set_cpu_params(self.estimator, num_estimator_cpus)
+
+        if feature_columns:
+            if isinstance(data, np.ndarray):
+                data = data[:, feature_columns]
+            else:
+                data = data[feature_columns]
+        with parallel_backend("ray", n_jobs=num_estimator_cpus):
```

Comment:
This will be set to `n_jobs=1` by default, which means it will fallback to joblib's SequentialBackend no matter what's set here. Unless the user specifically sets `num_estimator_cpus`, there will be no parallelism.

Reply:
Got it. Can we add a test for full batch inference with num_estimator_cpus greater than 1?